script.js

```diff
@@ -0,0 +1,194 @@
+    }
+    else if(curSortCriteria == 'popular') {
+        displayedRepoList.sort(
+            (b, a) => (a.stargazers_count + a.forks_count + a.watchers_count) - (b.stargazers_count + b.forks_count + b.watchers_count)
+        );
+    }
+    else if(curSortCriteria == 'original') {
+        displayedRepoList = [];
+        for(let k=0; k<repoList.length; k++) {
+            if(repoList[k].fork == false){
+                displayedRepoList.push(repoList[k]);
+            }
+        }
+        displayedRepoList.sort(
+            (b, a) => (a.stargazers_count + a.forks_count + a.watchers_count) - (b.stargazers_count + b.forks_count + b.watchers_count)
+        );
+    }
+    
+    let fragment = new DocumentFragment();
+    for(let i=0; i<displayedRepoList.length; i++){
+        
+        /* Create repo card */
+        let repoItem = document.createElement('div');
+        repoItem.classList.add('repo__card');
+        
+        /* Delete button */
+        let delButton = document.createElement('button');
+        delButton.innerHTML = '<i class="fas fa-times"></i>';
+        delButton.onclick = () => {
+            delButton.parentNode.style.maxHeight = '0';
+            delButton.parentNode.style.margin = '0';
+            setTimeout(() => {
+                delButton.parentNode.remove();
+            }, 500);
+        }
+        
+        /* Title of repo card */
+        let repoTitle = document.createElement('a');
+        repoTitle.href = displayedRepoList[i].html_url;
+        repoTitle.target = '_blank';
+        repoTitle.innerText = displayedRepoList[i].full_name;
+        
+        /* Details of repo */
+        let starsCount = document.createElement('div');
+        starsCount.innerHTML = '<i class="fas fa-star"></i> ';
+        starsCount.append(displayedRepoList[i].stargazers_count);
+        
+        let language = document.createElement('div');
+        language.innerHTML = '<i class="fas fa-code"></i> ';
+        language.append((displayedRepoList[i].language == null)?'--':displayedRepoList[i].language);
+        
+        let forksCount = document.createElement('div');
+        forksCount.innerHTML = '<i class="fas fa-code-branch"></i> ';
+        forksCount.append(displayedRepoList[i].forks_count);
+        
+        /* Avater of repo owner */
+        let repoAvatar = document.createElement('img');
+        repoAvatar.src = displayedRepoList[i].owner.avatar_url;
+        
+        
+        let repoDetails = document.createElement('div');
+        repoDetails.classList.add('repo__details');
+        repoDetails.append(language, starsCount, forksCount);
+        
+        let repoItemLeft = document.createElement('div');
+        repoItemLeft.append(repoTitle);
+        repoItemLeft.append(repoDetails);
+        
+        repoItem.append(repoAvatar);
+        repoItem.append(repoItemLeft);
+        repoItem.append(delButton);
+
+        fragment.append(repoItem);
```

Comment:
It's really hard for me to follow what's going on from viewing the code. I would recommend we use [templating string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) and [insertAdjacentHTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML) to construct the HTML. e.g:

```js
const name = 'John Doe';
const age = 3;

const htmlTemplate = `
   <ul>
      <li><b>Name:</b> ${name}</li>
      <li><b>Age:</b> ${age}</li>
   </ul>
`
document.getElementById("add_after_me").insertAdjacentHTML('afterend', htmlTemplate);
```